plenum/server/future_primaries_batch_handler.py

```diff
@@ -1,30 +1,75 @@
-import copy
-from collections import OrderedDict
-from typing import Tuple, Dict
+from typing import Dict, List
 
 from common.exceptions import LogicError
-from plenum.common.constants import TXN_TYPE, NODE, TARGET_NYM, ALIAS, SERVICES, VALIDATOR, DATA, POOL_LEDGER_ID
-from plenum.common.util import getMaxFailures
+from plenum.common.constants import POOL_LEDGER_ID, \
+    AUDIT_LEDGER_ID, AUDIT_TXN_PRIMARIES, AUDIT_TXN_VIEW_NO
+from plenum.common.ledger import Ledger
+from plenum.common.txn_util import get_payload_data
 from plenum.server.batch_handlers.batch_request_handler import BatchRequestHandler
 from plenum.server.batch_handlers.three_pc_batch import ThreePcBatch
-from plenum.server.pool_manager import TxnPoolManager
 
 
 class FuturePrimariesBatchHandler(BatchRequestHandler):
     # This class is needed for correct primaries storing in audit ledger.
 
     def __init__(self, database_manager, node):
         super().__init__(database_manager, POOL_LEDGER_ID)
+        # map of view_no and list of primaries
+        self.primaries = {}    # type: Dict[int, List]
+        self.db_manager = database_manager
         self.node = node
```

Comment:
It looks like we can remove the `node` dependency. Also it would be great to rename the class.

Reply:
Yes, after all of fixes regarding to primaries selection in view_change we can remove node's deps from it